File: python/path.py
import os
import sys
from pathlib import Path

# project_root_env = "PROJECT_DIR"
# project_root = os.getenv(project_root_env)
# if project_root == None:
#     raise EnvironmentError(f"{project_root_env} env variable is not set")
project_root = "/home/nvidia/git/mt"


def root(*argv) -> str: return os.path.join(project_root, *argv)
def res(*argv) -> str: return root(os.path.join("res", *argv))
def out(*argv) -> str: return root(os.path.join("out", *argv))
def ir(*argv) -> str: return res(os.path.join("ir", *argv))


def mkdir(path: str, parents=True, exist_ok=True): Path(path).mkdir(parents=parents, exist_ok=exist_ok)

File: cuda_fx_processor/src/utils/path.cpp
#include "path.hpp"
namespace path {

fs::path _workspace = fs::path("/home/nvidia/git/mt-dist");
fs::path _res = _workspace / "res";
fs::path _models = _res / "models";
fs::path _ir = _res / "ir";
fs::path _out = _workspace / "out";

std::string join(const std::string head, const std::string tail) {
    if (tail.empty()) {
        return head;
    } else {
        return fs::path(head) / tail;
    }
}

std::string join(const fs::path head, const std::string tail = "") {
    if (tail.empty()) {
        return head;
    } else {
        return head / tail;
    }
}

std::string workspace(const std::string path) {
    return join(_workspace, path);
}
std::string res(const std::string path) {
    return join(_res, path);
}
std::string models(const std::string path) {
    return join(_models, path);
}
std::string ir(const std::string path) {
    return join(_ir, path);
}
std::string out(const std::string path) {
    return join(_out, path);
}
};  // namespace path